drivers/pinctrl/pinctrl_silabs_dbus.c

```diff
@@ -19,16 +20,26 @@ int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt, uintp
 	for (uint8_t i = 0U; i < pin_cnt; i++) {
 		mem_addr_t enable_reg, route_reg;
 
+		/* Configure ABUS */
+		if (pins[i].en_bit == SILABS_PINCTRL_ANALOG) {
+			enable_reg = DT_INST_REG_ADDR_BY_NAME(0, abus) +
+				     (pins[i].base_offset * sizeof(mem_addr_t));
+			sys_write32(FIELD_PREP(ABUS_MASK(pins[i].mode), pins[i].route_offset),
+				    enable_reg);
+			continue;
+		}
```

Comment:
Coding style comments: 

  - The scope of the variables, the comments before the code blocks, the level of indentation, introduction of `ABUS_MASK()`, etc... makes me feel we probably want to introduce two sub functions for abus and for dbus.

  - I would rather write `0x7 << (pins[i].mode * 8)` rather than `GENMASK(...)` (I suggest to add `__ASSERT(pins[i].mode < 4)`).